### DISCUSSION TITLE
Silver Custom Realm: What are Client Scope Mappers? What should I keep in mind as I move to the Gold Standard Service?

### DISCUSSION THREAD

**Discussion by zsamji** [bcgov/sso-keycloak#153] [Gold Q&A]
I am using Client Scope Mappers in my custom realm - remind me what does this mean? If we move to Gold Standard Service, what should I keep in mind?

**Comment by tzhang200** [comment]
Client Scopes is a realm level concept, by that Pathfinder SSO team is researching on using client level hardcoded claim mapper to achieve Scope concept in client level. However, if you are using a mapper within your customized client scopes, that will add an additional level of complexity that SSO team needs to be aware of.

To find out if your customized Client Scope has Mappers setup in Keycloak:
![image](https://user-images.githubusercontent.com/3861078/182969922-8fc9d29f-1e1c-479a-bd7a-732fcd174a77.png)

1. select the client scope you want to review from the Client Scopes list

1. click on the Mappers tab. 

1. If a mapper is defined, click on the mapper, record down what Attribute and Token Claim Name are set up with.

1. Share with the SSO team who can work with you to determine if the information can be retrieved from some other existing attributes that gold standard realm provides.

More information on locating Client Scopes Mapper: https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/7.5/html-single/server_administration_guide/index#protocol
